File: docs/DSA/neetcode150/src/py/evaluate_reverse_polish_notation.py
class Solution:
    def evalRPN(self, tokens: list[str]) -> int:
        stk = []
        op = {"+", "-", "*", "/"}
        for token in tokens:
            if token not in op:
                stk.append(int(token))
            else:
                b= stk.pop()
                a = stk.pop()
                if token == "+":
                    stk.append(a+b)
                elif token == "-":
                    stk.append(a-b)
                elif token == "*":
                    stk.append(a*b)
                elif token == "/":
                    stk.append(int(a/b))
        return stk.pop()

File: DSA/src/go/evaluate_reverse_polish_notation.go
package main

import "strconv"

func evalRPN(tokens []string) int {
	var stk []int
	op := map[string]struct{}{
		"+": struct{}{},
		"-": struct{}{},
		"*": struct{}{},
		"/": struct{}{},
	}
	for _, token := range tokens {
		if _, ok := op[token]; !ok {
			token, _ := strconv.Atoi(token)
			stk = append(stk, token)
		} else {
			t2, t1 := stk[len(stk)-1], stk[len(stk)-2]
			var eval int
			if token == "+" {
				eval = t1 + t2
			} else if token == "-" {
				eval = t1 - t2
			} else if token == "*" {
				eval = t1 * t2
			} else if token == "/" {
				eval = t1 / t2
			}
			stk = append(stk[:len(stk)-2], eval)
		}
	}
	return stk[len(stk)-1]
}
